### PR TITLE
Time Plot Operator

### PR DESCRIPTION
**Pull request by jmaerte**
Added a time plot operator, that encapsulates the ntur40 plot. It relies on the time interval indirectly specified by the user and plots a numerical attribute of a given layer depending on the correlating temporal data.
Optional one can specify a grouping attribute, that colorizes the plot depending on the grouping data.

Maybe we should add a functionality in front-end that specifies whether or not a layer has temporal data, so we don't even bother offering to time plot this.

### CODE REVIEW

File: src/app/operators/types/timeplot-type.model.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+p = (
+  ${ggplot}
+  + geom_line()
+  + geom_point()
+  + scale_x_datetime(limits=query_lim_posix)
+  + xlab("Time") + ylab(\"${this.attribute}\")
+)
+print(p)
+`;
+        console.log(this.code);
```
Comment (ChristianBeilschmidt) [inline src/app/operators/types/timeplot-type.model.ts]: remove logging

**Comment by jmaerte** [comment]
I can't figure out the second point. Even though my test-datasets are not ordered by time axis values the final time-plot looks fine to me.

**Comment by jmaerte** [comment]
Travis: Build timed out because no output was received. I suggest a rebuild.

**Comment by ChristianBeilschmidt** [comment]
You still get an error when having no time range selected but just a single time stamp. Then you get the following error:
```
invalid request: OperatorException: HTTPService: query failed with error: OperatorException: R exception: Error evaluating: df = data.frame(start, attribute);
times = seq(mapping.qrect$t1, mapping.qrect$t2, (mapping.qrect$t2 - mapping.qrect$t1)/10)
p = (
  ggplot(df, aes(x=start,y=attribute))
  + geom_line()
  + geom_point()
  + scale_x_datetime(limits=query_lim_posix)
  + xlab("Time") + ylab("basisofrecord")
)
print(p)
}
```

Is there a specific reason why you divide the difference between `t2` and `t1` by 10?

The difference between `t2` and `t1` should be very small but not zero. Maybe this is a hint.

**Comment by ChristianBeilschmidt** [comment]
The check `mapping.qrect$t1 == mapping.qrect$t2` does not work since mapping always queries with a tiny, tiny query duration instead of a point in time.

Thus, I still get the same error.

Furthermore, I don't understand the change to the `time` variable (divided by 10 or not). In addition, I don't see it getting used anywhere.

**Comment by ChristianBeilschmidt** [comment]
Besides still having the problem that you cannot use the operator when you only use a time instance, you cannot reload the plot operator (page refresh) because there is no factory method defined.

**Comment by jmaerte** [comment]
Latest commit fixes the issue that just came up.
For the other problem: I cannot reproduce this. Maybe i misunderstood you. By saying "time instance" you mean that in the time config there is the toggle "same as start" set to true, right? Because in that case it gives me a default plot with the text "Dataset has no data in the specified time chunk". If you mean that the input set only has start time and no end time i have to take another look on this.

**Comment by ChristianBeilschmidt** [comment]
We somehow lost track of this PR. Can you merge with the current master? I will close this then.

**Comment by jmaerte** [comment]
Of course I will, but i still am not able to reproduce the open problem.

**Comment by ChristianBeilschmidt** [comment]
Travis-CI crashes with
`ERROR in src/app/operators/dialogs/time-plot-operator/time-plot-operator.component.html(2,26): : Expected 0 arguments, but got 1.`
Please check if it builds for debug as well as production.

**Pull request merged**